src/components/Fieldset/DynamicFieldset/ControlledDynamicFieldset.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import clsx from 'clsx';
+import '../Fieldset.scss';
+import type {ControlledDynamicFieldsetProps} from './DynamicFieldset.types';
+import {Switch, Typography} from '~/components';
+
+export const ControlledDynamicFieldset = React.forwardRef<HTMLDivElement, ControlledDynamicFieldsetProps>(({
+    id,
+    label,
+    helper,
+    children,
+    className,
+    buttons,
+    isOpen = false,
+    onChange,
+    ...props
+}, ref) => {
+    return (
+        <div
+            ref={ref}
+            id={id}
```

Comment:
You also have to add some `data-` or `aria-` attributes to match the state in order to be more semantic and help our tests with attributes to target